packages/sdk-btc/source/client.service.ts

```diff
@@ -1,16 +1,28 @@
-import { Contracts, Helpers, IoC, Services } from "@payvo/sdk";
+import { Collections, Contracts, Helpers, IoC, Services } from "@payvo/sdk";
 
 @IoC.injectable()
 export class ClientService extends Services.AbstractClientService {
 	public override async transaction(
 		id: string,
 		input?: Services.TransactionDetailInput,
 	): Promise<Contracts.ConfirmedTransactionData> {
-		return this.dataTransferObjectService.transaction(await this.#get(`transactions/${id}`));
+		const response = await this.#get(`transactions/${id}`);
+		return this.dataTransferObjectService.transaction(response.data);
+	}
+
+	public override async transactions(
+		query: Services.ClientTransactionsInput,
+	): Promise<Collections.ConfirmedTransactionDataCollection> {
+		let address = query.addresses ? query.addresses[0] : query.address;
```

Comment:
@faustbrian We currently don't support querying multiple addresses at the same time, but nothing prevents as from doing it... except for the endpoint `wallets/${address}/transactions` url (path param instead of query param). We'd need to change it to something like `wallets/transactions?addresses=${address}` or similar.

Reply:
This currently only exists because of ARK and probably won't have any use outside of that.

Reply:
@faustbrian Yet... I think we should do it for BTC. In the case of Ark it allows you to query multiple addresses (i.e. wallets) at the same time. But in the case of Bitcoin, one wallet is made of multiple addresses. So restoring a single wallet will be a pain otherwise (too many requests). So `/wallet/${address}/transactions` is actually not correct, because it will retrieve part of the wallet, not the whole thing. Same with balance. And same with utxos.
We need to think about this.

Reply:
Not much to think about if it has to be changed to be functional for how we'll implement the requests for those wallets. Change it to a `POST` endpoint that takes an array of addresses and we return all data for them with pagination.